# Activity - Multivariate LR 01

In [42]:
pip install word2number

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.


In [43]:
import pandas as pd
import numpy as np
from sklearn import linear_model
from word2number import w2n

In [44]:
df = pd.read_csv('./data/multi_var_act01.csv')
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0        NaN                    8.0                           9      50000
1        NaN                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

## Clean

In [45]:
df['experience'] = df['experience'].fillna('zero')
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0       zero                    8.0                           9      50000
1       zero                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

In [46]:
df['experience'] = df['experience'].apply(w2n.word_to_num)
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    NaN                           7      72000
7          11                    7.0                           8      80000

In [47]:
df['test_score(out of 10)'] = df['test_score(out of 10)'].fillna(df['test_score(out of 10)'].median())
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    8.0                           7      72000
7          11                    7.0                           8      80000

## Predict

In [48]:
reg = linear_model.LinearRegression()
reg.fit(df.drop('salary($)', axis='columns'), df['salary($)'])

LinearRegression()

In [49]:
reg.predict([[2, 9, 6]])

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([53205.96797671])

In [50]:
reg.predict([[12, 10, 10]])

C:\ProgramData\Anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([92002.18340611])

## Verify

In [51]:
reg.coef_

array([2812.95487627, 1845.70596798, 2205.24017467])

In [52]:
reg.intercept_

17737.263464337688

In [54]:
# 2 yr experience, 9 test score, 6 interview score
2812.95487627*2 + 1845.70596798*9 +  2205.24017467*6 + 17737.263464337688

53205.96797671769

In [55]:
# 12 yr experience, 10 test score, 10 interview score
2812.95487627*12 + 1845.70596798*10 +  2205.24017467*10 + 17737.263464337688

92002.18340607767

2 yr experience, 9 test score, 6 interview score = **53205.96797671**


12 yr experience, 10 test score, 10 interview score = **92002.18340611**